examples/bool.kl

```diff
@@ -0,0 +1,64 @@
+#lang "n-ary-app.kl"
+
+(import (shift "n-ary-app.kl" 1))
+(import (shift "quasiquote.kl" 1))
+
+(define not
+  (lambda (b)
+    (if b #false #true)))
+
+(define binary-and
+  (lambda (x y)
+    (if x y #false)))
+
+(define binary-or
+  (lambda (x y)
+    (if x #true y)))
+
+(define-macros
+  ((and (lambda (stx)
+          (syntax-case stx
+            ((cons _ args)
+             (syntax-case args
+               (()
+                (pure '#true))
+               ((cons x xs)
+                (pure `(binary-and ,x
+                                   ,(cons-list-syntax 'and xs stx)))))))))
+   (or (lambda (stx)
+         (syntax-case stx
+           ((cons _ args)
+            (syntax-case args
+              (()
+               (pure '#false))
+              ((cons x xs)
+               (pure `(binary-or ,x
+                                 ,(cons-list-syntax 'or xs stx)))))))))))
+
+(example (binary-and #false #false))
+(example (binary-and #false #true))
+(example (binary-and #true #false))
+(example (binary-and #true #true))
```

Comment:
I checked the results using `klister-repl run`, but I didn't bother adding examples to `test/Test.hs`, it's just so time-consuming. Maybe we're due for an `assert` macro?

Reply:
What about a "crash" primitive that terminates the program? `assert` can be nicely built on top of that.

Reply:
We already have `log` and `syntax-error`, I think that's good enough for writing `assert` in Klister

Reply:
I think `crash` needs to be `forall a . Syntax -> a` rather than `forall a . Syntax -> Macro a`